### PR TITLE
Trainer driver changes

### PR DESCRIPTION
**Pull request by goswaank**
1. Changed Predictor concrete classes, constructor calls the load method now. So remove any explicit load methods if called from outside and pass in modelDir parameter to the constructor
2. introduced config.ini which contains all the config
3. introduced another module: driver, which contains EmailTrainerDriver class which performs training on the CustomTrainer
4. changed httpserver.py to accommodate the above changes.

### CODE REVIEW

**Review by bedwards-clari — APPROVED** [review]
Looks good to me!

File: predictor/CustomModelPredictor.py [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
-        self.model = None
+        self.model = self.load(modelDir)
 
+    '''
+    Load the model named(best-model.pt) from the model directory passed in 
+    '''
     def load(self, modelDir):
-        self.model = SequenceTagger.load(modelDir + "best-model.pt")
+        if type(modelDir) == str:
+            path_to_file: Path = Path(modelDir+"best-model.pt")
```
Comment (dhruva-clari) [inline predictor/CustomModelPredictor.py:21]: Just checking: Is the file that we have to save in model_dir required to be named best-model.pt?

File: common/config.ini [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[DEFAULT]
+dataDir="~/hack22_resources/data/"
+modelDir="~/hack22_resources/model/"
+resourceDir="~/hack22_resources/resources/"
+inputFileName="email_data_full.json"
+
+[TRAINER]
+wordEmbeddingTypeStack=[WordEmbeddings('glove'),FlairEmbeddings('news-forward-fast'),FlairEmbeddings('news-backward-fast')]
```
Comment (IreneYangClari) [inline common/config.ini:8]: Just want to put a ping on this.. Later we can try different embedding without news content

**Review by dhruva-clari — APPROVED** [review]

File: common/config.ini (hunk 2 above)
Comment (goswaank) [reply common/config.ini:8]: absolutely Irene!

File: predictor/CustomModelPredictor.py (hunk 1 above)
Comment (goswaank) [reply predictor/CustomModelPredictor.py:21]: yes, flair automatically saves the best model with that name in the model directory provided.

File: driver/EmailTrainerDriver.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from flair.embeddings import *
+from common.DocumentEmbeddingTypes import DocumentEmbeddingTypes
+
+
+def main():
+
+    parser = ConfigParser()
+
+    ## read the config fuleConfig file
+    parser.read(os.path.expanduser('../common/config.ini'))
```
Comment (IreneYangClari) [inline driver/EmailTrainerDriver.py:14]: we don't need expanduser here. We can do `os.path.join(os.path.abspath(__file__), os.pardir, 'common/config.ini')`

File: webserver/httpserver.py [hunk 4]
```diff
@@ -63,21 +64,26 @@ def extract_query_params(self):
 # This is a stub scoring program. It will return 0 if the message
 # doesn't contain the word "win", and 100 if it does.
 def score_email(message, query_params):
-    predictor = get_predictor(query_params)
-    predictor.load("/home/ubuntu/hack22_resources/model")
+
+    ## Read the model directory from the ini file
+    parser = ConfigParser()
+    parser.read(os.path.expanduser('../common/config.ini'))
```
Comment (IreneYangClari) [inline webserver/httpserver.py:70]: we don't need expanduser here. We can do `os.path.join(os.path.abspath(__file__), os.pardir, 'common/config.ini')`

File: webserver/httpserver.py [hunk 5]
```diff
@@ -63,21 +64,26 @@ def extract_query_params(self):
...
+    model_dir = os.path.expanduser(eval(parser['DEFAULT']["modelDir"]))
+
+    predictor = get_predictor(query_params, model_dir)
     predicted_tuple = predictor.predict(predictor.preprocess(message))
     return predicted_tuple[0], int(predicted_tuple[1] * 100)
```
Comment (IreneYangClari) [inline webserver/httpserver.py:75]: since the predicted_tuple changed, we don't need index [0] here

File: driver/EmailTrainerDriver.py (hunk 3 above)
Comment (balasraj) [reply driver/EmailTrainerDriver.py:14]: good suggestion @IreneYangClari 👍

**Review by balasraj — APPROVED** [review]
Looks good to me, approving it..

**Review by IreneYangClari — APPROVED** [review]
Looks good.

File: driver/EmailTrainerDriver.py (hunk 3 above)
Comment (goswaank) [reply driver/EmailTrainerDriver.py]: Done with the change here!

File: webserver/httpserver.py (hunk 4 above)
Comment (goswaank) [reply webserver/httpserver.py]: Same, done with this change. please review the changes

**Pull request merged**